ovsdb/ovsdb-tool.c

```diff
@@ -1497,6 +1497,31 @@ do_check_cluster(struct ovs_cmdl_context *ctx)
         }
     }
 
+    /* Check for db consistency:
+     * The serverid must be in the servers list
+     */
+
+    for (struct server *s = c.servers; s < &c.servers[c.n_servers]; s++) {
+        struct shash *servers_obj = json_object(s->snap->servers);
+        const struct shash_node **nodes = shash_sort(servers_obj);
+        size_t n = shash_count(servers_obj);
+        char* server_id = xasprintf(SID_FMT, SID_ARGS(&s->header.sid));
+        bool found = false;
+        for (size_t i = 0; i < n; i++) {
+            const struct shash_node *node = nodes[i];
+            char *prev_serverid_entry = xasprintf("%.4s", node->name);
```

Comment:
We could avoid allocating memory for a new string and instead just do `!strncmp(node->name, prev_serverid_entry, SID_LEN)`.